src/nodes/display/ViewportDepthNode.js

```diff
@@ -116,6 +116,31 @@ export const viewZToPerspectiveDepth = ( viewZ, near, far ) => near.add( viewZ )
 // maps perspective depth in [ 0, 1 ] to viewZ
 export const perspectiveDepthToViewZ = ( depth, near, far ) => near.mul( far ).div( far.sub( near ).mul( depth ).sub( far ) );
 
+export const perspectiveDepthToLogarithmicDepth = ( perspectiveW, near, far ) => {
```

Comment:
In post processing we have to compute the view position of a fragment in various effects. We do that with `getViewPosition()`:

https://github.com/mrdoob/three.js/blob/9e5528d2890de5432c6472bde6702abecc8e1cf7/src/nodes/utils/PostProcessingUtils.js#L15

This method does currently not work if a logarithmic Depth buffer is used. How would the reverse operation of `perspectiveDepthToLogarithmicDepth()` look like that computes a "normal" depth value based on the logarithmic depth?

Reply:
@Mugen87 Computing the perspective `w` value from a logarithmic depth value using the new formula would look like this. Note that this gives you the `w` value, as in `modelViewProjection.w`. To get the "actual" `z` value, just divide `z` by `w` (I know you know that already but I'm just saying it for posterity).

![image](https://github.com/user-attachments/assets/78ae6f02-ea51-4a52-afcc-11320f74491e)

The TSL node function based on the math shown above would look like this. Like mentioned before, this returns the perspective `w` value, and not an "actual" depth. To get the actual depth, just do `z / w`:

```
const logarithmicDepthToPerspectiveDepth = (logDepth, near, far) => {
  // NOTE: this returns the "w" component, not "z"
  const exponent = logDepth.mul(log(far.div(near).add(1)));
  const e = 2.718281828459045235360287471352;
  return float(e).pow(exponent).sub(1).mul(near);
};
```

**NOTE: I have not tested the above node function above...it looks correct though...needs another set of eyes.**

As an aside, should the functions `perspectiveDepthToLogarithmicDepth()` and `logarithmicDepthToPerspectiveDepth()` be renamed to `perspectiveWToLogarithmicDepth()` and `logarithmicDepthToPerspectiveW()` to make it more clear that the `w` component is being used?

Desmos graph: https://www.desmos.com/calculator/1e3ttyfe6q

![image](https://github.com/user-attachments/assets/506a12bf-bd99-430c-a2b2-48022418792f)

Reply:
I've tried to replace the `sampleDepth()` function in `GTAONode` for quick testing but it does not work yet. Here is the code:
```js
const sampleDepth = ( uv ) => {

	const depth = this.depthNode.uv( uv ).x;

	if ( builder.renderer.logarithmicDepthBuffer === true ) {

		const mvp = positionWorld.mul( this.cameraProjectionMatrix );

		return mvp.z.div( logarithmicDepthToPerspectiveDepth( depth, this.cameraNear, this.cameraFar ) );

	} else {

		return depth;

	}

};
```
Do you see an obvious issue with the above code?

Reply:
I can say the following at the moment:

1. `depth` in `const depth = this.depthNode.uv( uv ).x;` should range between 0 and 1 before being passed to `logarithmicDepthToPerspectiveDepth( depth, this.cameraNear, this.cameraFar )`
2. Without looking at the rest of the code, `this.cameraNear` and `this.cameraFar` should be uniforms

In the meantime I will continue verifying that the `logarithmicDepthToPerspectiveDepth` is correct.

Reply:
Both requirements were true when during my tests. I'll revisit the code tomorrow.

Reply:
Does the final `depth` value returned from `sampleDepth` need to be linear/orthographic? If so, there should be a conversion from viewZ value you get after the division by `w` to orthographic:

```
const viewZ = mvp.z.div( logarithmicDepthToPerspectiveDepth( depth, this.cameraNear, this.cameraFar ) );

return viewZToOrthographicDepth( viewZ, this.cameraNear, this.cameraFar );

I am still confirming that the logarithmicDepthToPerspectiveDepth function is correct.
```

Reply:
> Does the final depth value returned from sampleDepth need to be linear/orthographic?

No, at least the AO uses the raw depth from the depth texture attachment.

Reply:
> As an aside, should the functions perspectiveDepthToLogarithmicDepth() and logarithmicDepthToPerspectiveDepth() be renamed to perspectiveWToLogarithmicDepth() and logarithmicDepthToPerspectiveW() to make it more clear that the w component is being used?

Yes, I think otherwise the names are confusing. Let's use `perspectiveWToLogarithmicDepth()`.

Reply:
I can't get it to work with the proposed formula. I've also tried a slightly different formula but with no success.
```js
export const logarithmicDepthToPerspectiveW = ( logarithmicW, near, far ) => {

	const logRatio = log2( far.div( near ).add( 1 ) );
	const perspectiveW = near.mul( pow( 2, logarithmicW.mul( logRatio ).sub( 1 ) ) );
	return perspectiveW;

};
```

Reply:
Perhaps your `mvp.z` equals (or is very close to) `mvp.w`? You may be returning `1` after dividing `z` by `w`.

Here is a fiddle where `mvp.z` equals `mvp.w`: https://jsfiddle.net/h7sd13pj

If that is the case, then try returning just the inverse:

`return float( 1 ).div( logarithmicDepthToPerspectiveDepth( depth, this.cameraNear, this.cameraFar ) );`

Reply:
Can you tell me how I can use `GTAONode` on my own to help you work on this? Maybe some little example or screenshot of what works when `logarithmicDepthBuffer` is `false` compared to when it is `true`?

Reply:
There is an example for demonstrating the AO: https://threejs.org/examples/webgpu_postprocessing_ao

I suggest you test this locally by turning on logarithmic depth buffer in the demo. You will see the AO breaks. 

For testing I have added  `logarithmicDepthToPerspectiveW()` to `ViewportDepthNode` and then tried to use it in `GTAONode` (with no success so far).